terrawrap/utils/tf_variables.py

```diff
@@ -30,15 +31,21 @@ def get_auto_vars(root_directory: str) -> Dict[str, Set[Variable]]:
                     # a variable is defined multiple times in the same time
                     # We will just use the last appearance of a variable
                     value = value[-1]
-                    if isinstance(value, list):
-                        value = tuple(value)
-                    if isinstance(value, dict):
-                        value = tuple(value.items())
-                    auto_vars[os.path.join(current_dir, file_name)].add(Variable(key, value))
+                    auto_vars[os.path.join(current_dir, file_name)].add(Variable(key, _make_hashable(value)))
 
     return dict(auto_vars)
 
 
+def _make_hashable(input_value):
+    if isinstance(input_value, list):
```

Comment:
are these the only data types we have to worry about for terraform variables?

Reply:
[Correct.](https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files)

Basically, you can only have the types that can be created in-line in Terraform, which is either a number, string, list, or dictionary.

Reply:
👍